.github/workflows/cmake-multi-platform.yml

```diff
@@ -0,0 +1,172 @@
+# Github CI config
+# Inspired by https://github.com/libcpr/cpr/blob/master/.github/workflows/ci.yml
+# Runs: clang format check, g++ & clang, release & debug, on Fedora
+
+name: OSTree-TUI CI
+
+on: [push, workflow_dispatch, pull_request]
+
+env:
+  # standard dependencies for building & project
+  BUILD_DEPENDENCIES: "gcc-c++ clang git gcc gdb make cmake automake autoconf"
+  OSTREE_TUI_DEPENDENCIES: "glib2-devel gtk-doc ostree-devel libtool bison liblzf e2fsprogs-devel xz-devel gpgme-devel fuse-devel"
```

Comment:
Does it make sense to further optimise the build process to create a custom docker/container image based on fedora:latest where those are already installed?

This docker image should then be rebuild automatically e.g. every week once so it stays up to date. As far as I'm aware we could host this image inside the GitHub container registry then: https://github.blog/news-insights/product-news/introducing-github-container-registry/

This is nothing that needs to happen now, but could be a followup PR to improve CI speed.

Reply:
It does make sense. I already had something like that in mind, but I didn't look into it, yet. Definitely something I will integrate in a follow up :)

Reply:
Perfect. Can you please create an issue for it?